src/main/java/com/byoutline/eventcallback/RetrofitResponseEvent.java

```diff
@@ -0,0 +1,15 @@
+package com.byoutline.eventcallback;
+
+import retrofit.client.Header;
+
+import java.util.List;
+
+/**
+ * Event that will have server response body, headers and status.
+ */
+public interface RetrofitResponseEvent<R> extends ResponseEvent<R> {
+
+    void setHeaders(List<Header> headers);
+
+    void setStatus(int status);
```

Comment:
It is probably worth considering using single setter with multiple arguments instead. @smdremedy What's your take on this?